test/Test/WebDriver/WebDriverTest.php

```diff
@@ -53,16 +61,37 @@ protected function tearDown()
     }
 
     /**
+     * Returns the full url to the test site (corresponding to the root dir of the library).
+     * You can set this via env var ROOT_URL
+     * @return string
+     */
+    protected function getTestDocumentRootUrl()
+    {
+        return $this->testDocumentRootUrl;
+    }
+
+    /**
+     * Returns the full url to the Selenium server used for functional tests
+     * @return string
+     *
+     * @todo make this configurable via env var
+     */
+    protected function getTestSeleniumRootUrl()
+    {
+        return $this->testSeleniumRootUrl;
+    }
+
+    /**
      * @group Functional
      */
     public function testSessions()
     {
         try {
-	    $this->assertCount(0, $this->driver->sessions());
+        $this->assertCount(0, $this->driver->sessions());
 
             $this->session = $this->driver->session();
         } catch (\Exception $e) {
-            if (strpos($e->getMessage(),'Failed connect to localhost:4444; Connection refused') !== false
+            if (preg_match('/Failed to connect to .* Connection refused/', $e->getMessage()) != false
```

Comment:
1. no idea. I left it as I found it
2. agree. I just did not want to change too much stuff